### ISSUE TITLE
Improve type clarity of twisted.conch.ssh.transport

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8237]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8237|
|Type|enhancement|
|Created|2016-03-18 08:47:11Z|
|Branch|https://github.com/twisted/twisted/tree/clarity-sshtransport-8237|

Change string literals to be bytes literals, that sort of thing, to prepare for porting.

<details><summary>Searchable metadata</summary>

```
trac-id__8237 8237
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__branches_clarity_sshtransport_8237 branches/clarity-sshtransport-8237
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__None None
time__1458290831136308 1458290831136308
changetime__1458522276015219 1458522276015219
version__None None
owner__hawkowl hawkowl
cc__z3p
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

(In [47018]) Branching to clarity-sshtransport-8237.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

Changes made, builders spun, please review.

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)|@Lukasa set owner to @Lukasa<br>@Lukasa set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_5"></a>|@Lukasa set owner to @hawkowl<br>@Lukasa set status to `new`|
|-|-|

Generally this looks fine, and I have nothing that should block a merge. I should note that this is not a complete Python 3 port: the tests *will* fail when run on Python 3. However, this ticket doesn't pretend to be a full port, so that's fine. Some notes:

- `transport.py`, lines 547-557: that's an awful lot of string concatenation happening there, even with all those joins. Are we sure that there's nothing to be gained from adding an outer `b''.join()` there? Also...what's up with the octal literals?
- `transport.py`, lines 644-645 and 652-655: those strings probably need to be bytestrings too.
- `test_transport.py`, line 647: This won't work. On Python 3, chr() returns a unicode string, but packet is a bytestring. This test will end up failing. This pattern (use of chr()) occurs in a couple of other places in this test (e.g. L963) that might want to get fixed up while we're here, though you can also choose to leave them for a later patch.

Otherwise, LGTM. Builds are green, and you can merge to trunk as is if you like.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl set status to `closed`|
|-|-|

(In [47034]) Merge clarity-sshtransport-8237: Improve clarity of types in twisted.conch.ssh.transport

Author: hawkowl
Reviewer: lukasa
Fixes: [#8237](https://github.com/twisted/twisted/issues/8237)
